### PR TITLE
RET-3460-Final - Refactored GenericTypeItem to TypeItem

### PR DESCRIPTION
**Pull request by aaronjduffy**
https://tools.hmcts.net/jira/browse/RET-3460

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [x] Does this PR introduce a breaking change

### CODE REVIEW

**Review by lvo700 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelper.java [hunk 1]
```diff
@@ -779,8 +778,8 @@ private static StringBuilder getAddressLabelsData(int numberOfCopies, int starti
         return sb;
     }
 
-    private static int getPageLabelNumber(int startingLabel, int i) {
-        int pageLabelNumber = i + 1;
+    private static int getPageLabelNumber(int startingLabel, int currentNumber) {
+        int pageLabelNumber = currentNumber + 1;
```
Comment (marcjudge) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelper.java:782]: Use a constant rather than 1

File: build.gradle [hunk 2]
```diff
@@ -336,7 +336,7 @@ dependencyManagement {
 dependencies {
 
     implementation group: 'com.github.hmcts', name: 'et-common', version:'1.4.0'
-    implementation group: 'com.github.hmcts', name: 'et-data-model', version:'3.3.1'
+    implementation group: 'com.github.hmcts', name: 'et-data-model', version:'RET-3460-Final-SNAPSHOT'
```
Comment (marcjudge) [inline build.gradle:339]: Update

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelper.java (hunk 1 above)
Comment (lvo700) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/DocumentHelper.java]: Refactored

File: build.gradle (hunk 2 above)
Comment (lvo700) [reply build.gradle:339]: I think this needs to stay until it has been QA'd as they won't be able to verify it passes pipeline otherwise.

**Pull request closed without merging**